sw/device/lib/dif/dif_sensor_ctrl.h

```diff
@@ -12,13 +12,129 @@
  */
 
 #include <stdint.h>
-
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
 #include "sw/device/lib/dif/autogen/dif_sensor_ctrl_autogen.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif  // __cplusplus
 
+/**
+ * An event, identified by a numeric id.
+ *
+ */
+typedef uint32_t dif_sensor_ctrl_event_idx_t;
+
+/**
+ * A vector of AST events, where each bit represents one event.
+ *
+ */
+typedef uint32_t dif_sensor_ctrl_events_t;
+
+/**
+ * IO power status
+ *
+ */
+typedef uint32_t dif_sensor_ctrl_io_power_status_t;
+
+/**
+ * Locks sensor control configuration from further updates.
+ *
+ *
+ * @param sensor_ctrl A sensor_ctrl handle.
+ * @return 'kDifBadArg' if `handle` is null.
+ */
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_sensor_ctrl_lock_cfg(const dif_sensor_ctrl_t *sensor_ctrl);
+
+/**
+ * Sets the value for a partciular AST event trigger.
+ *
+ * @param sensor_ctrl A sensor_ctrl handle.
```

Comment:
`event_idx` is feeling left out with no documentation :smile:

Reply:
updated! also changed to `event_idx` where it should be